pkgs/stdenv/generic/setup.sh

```diff
@@ -174,6 +199,63 @@ addToSearchPath() {
     addToSearchPathWithCustomDelimiter ":" "$@"
 }
 
+# Prepend elements to variable "$1", which may come from an attr.
+#
+# This is useful in generic setup code, which must (for now) support
+# both derivations with and without __structuredAttrs true, so the
+# variable may be an array or a space-separated string.
+#
+# Expressions for individual packages should simply switch to array
+# syntax when they switch to setting __structuredAttrs = true.
+_prepend() {
```

Comment:
I don't think these functions should be considered "private", as suggested by the underscore, because other shared code needs them too. Setup hooks come to mind.

Could you remove the underscore and add them to `ssec-stdenv-functions` in the manual?

Reply:
How about renaming them to
prependToVar
appendToVar